app/src/main/kotlin/io/sweers/catchup/app/ApplicationModule.kt

```diff
@@ -139,5 +151,88 @@ abstract class ApplicationModule {
       // This makes a call to disk, so initialize it off the main thread first... ironically
       Dispatchers.Main
     }
+
+    @Initializers
+    @JvmStatic
+    @IntoSet
+    @Provides
+    fun coilInit(imageLoader: ImageLoader): () -> Unit = {
+      Coil.setDefaultImageLoader(imageLoader)
+      CoilLogger.setEnabled(BuildConfig.DEBUG)
+    }
+
+    @Qualifier
+    @Retention(BINARY)
+    private annotation class CoilOkHttpStack
+
+    @Qualifier
+    @Retention(BINARY)
+    annotation class IsLowRamDevice
+
+    @IsLowRamDevice
+    @Singleton
+    @JvmStatic
+    @Provides
+    fun isLowRam(@ApplicationContext context: Context): Boolean {
+      // Prefer higher quality images unless we're on a low RAM device
+      return context.getSystemService<ActivityManager>()?.let {
+        ActivityManagerCompat.isLowRamDevice(it)
+      } ?: true
+    }
+
+    @CoilOkHttpStack
+    @Singleton
+    @JvmStatic
+    @Provides
+    fun coilCache(@ApplicationContext context: Context): Cache = createDefaultCache(context)
+
+    @CoilOkHttpStack
+    @Singleton
+    @JvmStatic
+    @Provides
+    fun coilHttpClient(
+      okHttpClient: OkHttpClient,
+      @CoilOkHttpStack cache: Cache
+    ): OkHttpClient {
+      return okHttpClient.newBuilder()
+          .cache(cache)
+          .build()
+    }
+
+    @Singleton
+    @JvmStatic
+    @Provides
+    fun imageLoader(
+      @ApplicationContext context: Context,
+      @IsLowRamDevice isLowRamDevice: Boolean,
+      @CoilOkHttpStack okHttpClient: dagger.Lazy<OkHttpClient>
+    ): ImageLoader {
+      return ImageLoader(context) {
+        // Coil will do lazy delegation on its own under the hood, but we
+        // don't need that here because we've already made it lazy. Wish this
+        // wasn't the default.
+        callFactory(object : Call.Factory {
+          override fun newCall(request: Request): Call {
+            return okHttpClient.get().newCall(request)
+          }
+        })
+
+        // Hardware bitmaps don't work with the saturation effect
+        allowHardware(false)
+        allowRgb565(isLowRamDevice)
+        crossfade(true)
```

Comment:
`crossfade(true)` is an alias for `crossfade(100)` (100 is the default duration), similarly `crossfade(false)` = `crossfade(0)`

Reply:
Good to know! Will clean this up